### DISCUSSION TITLE
Read variables from GCP secret manager secret backend for airflow

### DISCUSSION THREAD

**Discussion by rgarg-quotient** [apache/airflow#36087] [Q&A] [answered]
Hi,

I have configured secret backend to GCP secret manager following document: https://airflow.apache.org/docs/apache-airflow-providers-google/stable/secrets-backends/google-cloud-secret-manager-backend.html 

Post setup I am able to fetch secret value using command:
```airflow variables get <<variable-name>>```

Now I want to read secret stored in GCP secret manager as a variable in a DAG. I didn't find any documentation around it. Please guide on how to do this. If any documentation link will be helpful.

**Comment by potiuk** [comment, marked as answer]
Just read them as usual https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/variables.html#variables (providing that you have them configured in your DAGs an tasks). Just make sure DO NOT use them at top-level of the DAG code as our best practices say https://airflow.apache.org/docs/apache-airflow/stable/best-practices.html#top-level-python-code 

Generally also look through other docs. Airlfow docs have search and it usually hepful in finiding answers.
